openedx-unsupported/configuration
# Adds ANALYTICS_API_AGGREGATE_PAGE_SIZE, default value 10

Adds a new `ANALYTICS_API_AGGREGATE_PAGE_SIZE` variable, with default value `10`.

**JIRA tickets**: [OSPR-1695](https://openedx.atlassian.net/browse/OSPR-1695)

**Discussions**: Please see https://github.com/edx/edx-analytics-data-api/pull/162#issuecomment-284547729

**Partner information**: 3rd party-hosted open edX instance

**Deployment targets**: insights.edx.org and insights.edge.edx.org

**Merge deadline**: 31 Mar 2017 if possible.  We'd like to deploy this change when upgrading our clients to Ficus.

**Testing instructions**:

1. Create an [analytics devstack](http://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/installation/analytics/install_analytics.html), using this branch as your `configuration_version`
1. Ensure that the new `ANALYTICS_API_AGGREGATE_PAGE_SIZE: 10` shows up in your `/edx/etc/analytics_api.yml` file.

**Author notes and concerns**:

1. The [elasticsearch_dsl aggregate search results defaults to 10](https://github.com/elastic/elasticsearch-dsl-py/blob/1a9092429ac582c0cf33f8d6287a20266be063cd/elasticsearch_dsl/search.py#L261), so that's why we're using `10` as the default.  But would be fine with anything you think is reasonable.

**Reviewers**
- [x] @edx/devops reviewers

Configuration Pull Request
---

Make sure that the following steps are done before merging

  - [-] are you adding any new default values that need to be overridden when this goes live? No.
